pkg/kubelet/prober/prober_manager.go

```diff
@@ -253,8 +246,15 @@ func (m *manager) UpdatePodStatus(podUID types.UID, podStatus *v1.PodStatus) {
 				ready = result == results.Success
 			} else {
 				// The check whether there is a probe which hasn't run yet.
-				_, exists := m.getWorker(podUID, c.Name, readiness)
-				ready = !exists
+				w, exists := m.getWorker(podUID, c.Name, readiness)
+				ready = !exists // no readinessProbe -> always ready
+				if exists {
+					// Trigger an immediate run of the readinessProbe to update ready state
+					select {
+					case w.manualTriggerCh <- struct{}{}:
+					default: // Non-blocking.
```

Comment:
add logging that we failed to manually trigger the update?

Reply:
I don't know... does it work putting a log under `default`?

Reply:
Hard to test, but it does: https://gobyexample.com/non-blocking-channel-operations#non-blocking-channel-operations:~:text=Therefore%20the%20default%20case%20is%20selected.